test/ContentValidationListenerTest.php

```diff
@@ -2570,4 +2968,51 @@ public function testDeleteRequestWithBody()
         $this->assertNotContains('Value is required and can\'t be empty', $error_messages['bar']);
         $this->assertContains('The input does not match against pattern \'/^[a-z]+/i\'', $error_messages['bar']);
     }
+
+    public function testReturnsNothingOnDeleteRequestIfContentIsInValidAndValidationSetViaInputFilterKeyword()
+    {
+        $services = new ServiceManager();
+        $factory = new InputFilterFactory();
+        $services->setService(
+            'FooValidator',
+            $factory->createInputFilter(
+                [
+                'foo' => [
+                    'name' => 'foo',
+                    'validators' => [
+                        ['name' => 'Digits'],
+                    ],
+                ],
+                ]
+            )
+        );
+        $listener = new ContentValidationListener(
+            [
+                'Foo' => ['input_filter' => 'FooValidator'],
+            ],
+            $services
+        );
+
+        $request = new HttpRequest();
+        $request->setMethod('DELETE');
+
+        $matches = $this->createRouteMatch(['controller' => 'Foo']);
+
+        $dataParams = new ParameterDataContainer();
+        $dataParams->setBodyParams(
+            [
+                'foo' => 'not digit data',
+            ]
+        );
+
+        $event = new MvcEvent();
+        $event->setName('route');
+        $event->setRequest($request);
+        $event->setRouteMatch($matches);
+        $event->setParam('ZFContentNegotiationParameterData', $dataParams);
+
+        $this->assertNull($listener->onRoute($event));
+        $this->assertNull($event->getResponse());
+        return $event;
```

Comment:
Why we return it here? Where it is used?

Reply:
No, I am not using this, but I added it for future if someone will need to have dependency with this function. 
But I will remove it currently to keep code more clear.